MonoGame.Framework/Random.cs

```diff
@@ -0,0 +1,386 @@
+                }
+            }
+            
+            /// <summary>
+            /// Read-only variable containing the current state of the random number generator internally.
+            /// This variable can be assigned as the seed of another random number generator to make them
+            /// produce the same output, or can be used to save the state of the RNG.
+            /// </summary>
+            [CLSCompliant(false)]
+            public ulong State
+            {
+                get
+                {
+                    return this._state;
+                }
+            }
+
+            /// <summary>
+            /// Non-negative number identifying the sequence of random numbers produced by the generator.
+            /// </summary>
+            /// <devdoc>The sequence number determines the increment of the internal LCG. Only uneven,
+            /// unsigned values are valid. Allowing any positive long value is more intuitive.</devdoc>
+            public long Sequence
+            {
+                get
+                {
+                    return (long)(_seq >> 1);
+                }
+                set
+                {
+                    if (value < 0)
+                        throw new ArgumentOutOfRangeException();
+                    this._seq = ((ulong)value << 1) | 1;
+                }
+            }
+
+            #endregion
+
+            #region Constructors
+
+            /// <summary>
+            /// Initializes a new instance of the random number generator using the value of 
+            /// Random.GenerateSeed() as a default seed value and the default sequence.
+            /// </summary>
+            /// <devdoc>The default sequence is derived from Knuth's LCG for MMIX. When fed into the
+            /// Sequence property, it produces an increment of 1442695040888963407.</devdoc>
+            public Random()
+                : this(GenerateSeed(), 721347520444481703)
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the random number generator using the value of
+            /// Random.GenerateSeed() as a default seed value and the specified sequence.
+            /// </summary>
+            /// <param name="sequence">Non-negative number identifying the sequence of random numbers.</param>
+            public Random(long sequence)
+                : this(GenerateSeed(), sequence)
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the random number generator using the specified seed and sequence.
+            /// </summary>
+            /// <param name="seed">Seed to be used when computing new random numbers</param>
+            /// <param name="sequence">Non-negative number identifying the sequence of random numbers.</param>
+            [CLSCompliant(false)]
+            public Random(ulong seed, long sequence)
+            {
+                this.Seed = seed;
```

Comment:
If `sequence` should be non-negative should we be asserting or throwing on a bad input here?